### ISSUE TITLE
Support for terminal bell as a notification

### ISSUE THREAD

**Issue by mozzwald** [termux/termux-app#115]
The terminal bell works when the app is running in the foreground with the screen on but after screen turns off or the app is running in the background it no longer plays a sound or vibrates based on the _bell-character_ setting in _termux.properties_.

This option is useful for being notified of events if connected to another machine via ssh (where termux-notification will not work). Please add support for terminal bell as a notification.

**Comment by mozzwald** [comment]
As a work around I'm using tmux v2.2 to handle the terminal bell in termux with the following:

`apt install tmux play-audio`

usr/etc/profile.d/tmux.sh
```
if [ $TMUX ] ; then
        tmux set-hook -g alert-bell 'run "notif.sh ; return 0"'
fi
```
usr/bin/notif.sh
```
#!/data/data/com.termux/files/usr/bin/sh
# enable terminal bell via tmux

termux-notification -i 4848 -t "Termux Terminal Bell" -c "Termux Terminal Bell Notification" &
play-audio -s notification /system/media/audio/notifications/Xenon.ogg &
termux-vibrate -d 150 &
```
The profile script is another workaround for tmux 2.2 that will be fixed in 2.3 (already fixed in tmux master branch). In tmux v2.3, the profile script will not be needed and the set-hook option can be added directly to your tmux.conf file. The notif.sh script can be modified to your liking and is run whenever there is a terminal bell.

**Comment by twaik** [comment]
Is this still relevant? I can not make system play system bell sound on both Termux and Konsole (I tried `echo -en '\a'`, `echo ^G`, `tput bel`, nothing works).

**Comment by TomJo2000** [comment]
> Is this still relevant? I can not make system play system bell sound on both Termux and Konsole (I tried `echo -en '\a'`, `echo ^G`, `tput bel`, nothing works).

@twaik 
`printf '\a'`, `echo '\a'` and `tput bel` output a [BEL](https://en.wikipedia.org/wiki/Bell_character) control character to the terminal.
`echo ^G` does not, the GNU Bash or POSIX version of `echo` does not interpret [caret notation](https://en.wikipedia.org/wiki/Caret_notation).

A BEL char being printed to the terminal causes Termux to send a short vibration to Android on my Device (Android 10), check if you have vibration enabled in the Android quick settings.

**Comment by twaik** [comment]
> A BEL char being printed to the terminal causes Termux to send a short vibration to Android on my Device (Android 10), check if you have vibration enabled in the Android quick settings.

You are right. I succeed making constantly vibrating phone with `while :; do tput bel; done`.

**Comment by TomJo2000** [comment]
There is actually a dedicated escape sequence for notifications,
it's a `urxvt` escape sequence extension though, `OSC 777`.
i.e `\e]777;notify;<title>;<body>\e\\`
- See: https://codeberg.org/dnkl/foot/pulls/236

I don't think Termux implements it.
But it is implemented by: `urxvt`(obviously), `foot` and `GNOME terminal/libvte`

**Comment by jccbbb** [comment]
Would also find this extremely useful 🎉
